lib/stud/task.rb

```diff
@@ -36,5 +36,14 @@ def stop!
     def self.interrupted?
       Thread.current[:stud_task_interrupted]
```

Comment:
@ph the `Task.interrupted?` class method will not access the same threadlocal variable as the `task.stop!` instance method.  `Task.interrupted?` will set the `:stud_task_interrupted` theadlocal of the caller thread. 

either we use an object instance variable of the `Task` class, ie `@stud_task_interrupted`, or we use the `@thread[:stud_task_interrupted]` everywhere...

haven't analyzed why a `Task` class `@stud_task_interrupted` ivar would not work here... but looking quickly at implementation, a `Task` object can only hold a single thread at a time and executes only once so I don't think we need a threadlocal here...